### ISSUE TITLE
error NotReadableError Could not start video source

### ISSUE THREAD

**Issue by chitainguyen** [JodusNodus/react-qr-reader#137]
Dear,
It throw error NotReadableError when i turn off screen and reopen it, QR Reader can not work after that.

**Comment by pritamInnoraft** [comment]
This issue still exists on the Samsung s20+ and Samsung s10+.
Try : [Try here](https://codesandbox.io/s/jolly-currying-xrqci?file=/src/App.js)

**Comment by perfectmak** [comment]
This is also happening in chrome on Samsung s20

**Comment by kangil-kim** [comment]
This doesn't work in chrome on Samsung s10 but work in firefox

**Comment by perfectmak** [comment]
Yeah, I had to fork the repository and use the fixes in the #154 PR to get it to work on Samsung devices.

**Comment by kangil-kim** [comment]
@perfectmak Did you deploy to npm server? I want to check on codesandbox.io

**Comment by perfectmak** [comment]
@kangil-kim I adapted it for my specific use case and published it as https://www.npmjs.com/package/eth-qr-scanner.

**Comment by kangil-kim** [comment]
@perfectmak your package works on s10 5g ang s20, but I think, handleScan is not working. thanks by the way.

**Comment by perfectmak** [comment]
@kangil-kim Yes, I modified my package such that the handleScan only gets triggered if the QR code contains a valid Ethereum format address. So it doesn't work for just any QR Code.

**Comment by kangil-kim** [comment]
@perfectmak oh understood. really thanks :)

**Comment by kangil-kim** [comment]
@perfectmak facingMode={'environment'} is not working on ios. Do you know why?

**Comment by perfectmak** [comment]
@kangil-kim I don't know. I think it is a bug, because I noticed that is you try multiple times with environment, it would eventually open the back camera

**Comment by kangil-kim** [comment]
@perfectmak 

I fixed and using IOS `environment` facing mode. but I don't know why `getDeviceId` function doesn't work on IOS.

```
const isIos = navigator.userAgent.match(/iPhone|iPad|iPod/i);
const vConstraintsPromise = isIos
        ? Promise.resolve(props.constraints || constraints)
        : getDeviceId(props.facingMode).then(deviceId => Object.assign({}, { deviceId }, props.constraints))
```

**Comment by perfectmak** [comment]
@kangil-kim It's nice you got it to work. I don't have an iOS device so can't really help with that scenario.
